src/LiveQuery/QueryTools.js

```diff
@@ -206,7 +206,9 @@ function matchesKeyConstraints(object, key, constraints) {
         }
         break;
       case '$exists':
-        if (typeof object[key] === 'undefined') {
+        let propertyExists = typeof object[key] !== 'undefined';
+        let existenceIsRequired = constraints['$exists'];
```

Comment:
Hum.. I see. Should I enclose it under an `if (typeof constraints['$exists']  === "boolean")`?